main.go

```diff
@@ -60,12 +62,21 @@ func main() {
 }
 
 func pkgsInfo(r io.Reader, cache map[string]string) error {
+	data, err := io.ReadAll(r)
```

Comment:
Can  you add an `io.LimitReader` here? In the scanner version we go line by line so I don't care much about size.

Reply:
Make sense. Per the module proxy doc, go.mod files are limited to 16 MiB actually.
https://go.dev/ref/mod#zip-path-size-constraints